Bang/GameEvents/CharacterEffects/States/WaitingForCardSelectionState.cs

```diff
@@ -13,6 +13,12 @@ public WaitingForCardSelectionState(Gameplay gameplay) : base(gameplay)
 
         public override HandlerState ApplyCardEffect(Player player, BangGameCard card)
         {
+            if (!((ChooseCardsResponse)SideEffect).CardsToChoose.Contains(card))
+            {
+                Logger.Error("Kit Karlson can't drop card from his hand during phase one");
+                return this;
```

Comment:
I think it would be nicer if we return `ErrorState` with `NotAllowedOperation` as side effect.